### PR TITLE
Sort messages by id

### PR DESCRIPTION
**Pull request by Cryptex-github**

### CODE REVIEW

File: ferrischat_webserver/src/messages/message_history.rs [hunk 1]
```diff
@@ -19,6 +19,8 @@ pub async fn get_message_history(
 
     let mut limit = params.limit;
 
+    let oldest_first = params.oldest_first;
```
Comment (tazz4843) [inline ferrischat_webserver/src/messages/message_history.rs:22]: Just unpack the struct here instead.
```rs
let GetMessageHistoryParams { mut limit, oldest_first } = params;
```

File: ferrischat_webserver/src/messages/message_history.rs [hunk 2]
```diff
@@ -30,16 +32,22 @@ pub async fn get_message_history(
...
 
-    let messages = {
-        let resp = sqlx::query!(
-            r#"SELECT m.*, (
+    let mut query: &str = "";
+
+    if oldest_first == Some(true) {
+        query = r#"SELECT m.*, (
+                SELECT u.* FROM users u WHERE id = m.author_id
+            ) AS author FROM messages m WHERE channel_id = $1 ORDER BY id AESC LIMIT $2"#;
```
Comment (jay3332) [inline ferrischat_webserver/src/messages/message_history.rs:40]: should be ``ASC`` not ``AESC``

File: ferrischat_webserver/src/messages/message_history.rs [hunk 3]
```diff
@@ -30,16 +32,22 @@ pub async fn get_message_history(
...
+    } else {
+        query = r#"SELECT m.*, (
                 SELECT u.* FROM users u WHERE id = m.author_id
-            ) AS author FROM messages m WHERE channel_id = $1 LIMIT $2"#,
-            bigint_channel_id,
-            limit
-        )
-        .fetch_all(db)
-        .await;
+            ) AS author FROM messages m WHERE channel_id = $1 ORDER BY id AESC LIMIT $2"#
```
Comment (jay3332) [inline ferrischat_webserver/src/messages/message_history.rs:44]: should be ``ASC`` not ``AESC``

File: ferrischat_webserver/src/messages/message_history.rs [hunk 4]
```diff
@@ -30,16 +32,22 @@ pub async fn get_message_history(
...
+    }
+
+    let messages = {
+        let resp = sqlx::query!(query, bigint_channel_id, limit)
```
Comment (hydrostaticcog) [inline ferrischat_webserver/src/messages/message_history.rs:47]: `sqlx::query` expects a string literal; will fail to compile.

**Review by hydrostaticcog — CHANGES_REQUESTED** [review]

**Review by tazz4843 — APPROVED** [review]

**Pull request merged**